build-system/tasks/performance/measure-documents.js

```diff
@@ -118,6 +196,124 @@ const readMetrics = (page) =>
     };
   });
 
+/**
+ * Set up defaults handlers for docs that will be requested
+ * that are not explicted stated as script tags and not
+ * handled by the 'additionalHandlers'.
+ *
+ * @param {!Array<function>} handlersList
+ * @param {string} version
+ */
+function setupDefaultHandlers(handlersList, version) {
+  // Handle requests made by runtime for default extensions
+  handlersList.push((interceptedRequest) =>
+    defaultExtensionsHandler(interceptedRequest, version)
+  );
+}
+
+/**
+ * Set up appropriate handlers based upon handlerOptions
+ *
+ * @param {!Array<function>} handlersList
+ * @param {?Object} handlerOptions
+ * @param {!Puppeteer.page} page
+ * @param {!function} resolve
+ */
+async function setupAdditionalHandlers(
+  handlersList,
+  handlerOptions,
+  page,
+  resolve
+) {
+  switch (handlerOptions.handlerName) {
+    case 'analyticsHandler':
+      // Set up timing
+      Object.assign(handlerOptions, {'startTime': Date.now()});
+      handlersList.push((interceptedRequest) =>
+        handleAnalyticsRequests(
+          interceptedRequest,
+          Object.keys(handlerOptions.extraUrlParam)
+            .map((key) => `${key}=${handlerOptions.extraUrlParam[key]}`)
+            .toString(),
+          (endTime) => {
+            if (!handlerOptions.endTime) {
+              Object.assign(handlerOptions, {
+                endTime,
+                'timeout': 0,
```

Comment:
To confirm you've tried with multiple urls, and they work fine? Asking again because my understanding is this would override the handlerOptions['timeout'] to 0, and would affect the next test. 
There's could be a deep copy somewhere that I missed.

Reply:
Yup tried it multiple urls. Each URL has a copy of the handlerOptions so no worry about mistakenly overriding something.

Reply:
Thanks for confirming